lib/filters/http-headers.js

```diff
@@ -20,6 +20,11 @@ function httpHeaders (obj) {
 
```

Comment:
This is weird one. Because of the way our pre-existing header **redacting** (not filtering) interacted with the descision to use `undefined`, without this change out `undefined` header would be changed to have a values of `REDACTED`, and would not be removed from the payload.  This change fixes that.